src/plugins/intel_gna/src/transformations/insert_identity_layer.hpp

```diff
@@ -68,6 +68,73 @@ class IdentityCandidatesCleanup : public ov::pass::ModelPass {
     bool run_on_model(const std::shared_ptr<ov::Model>& m) override;
 };
 
+/**
+ * @brief Inserts identity for precision agnostic (or FQ) concat inputs
+ * Example model:
+ *
+ *                      Parameter
+ *                          |
+ *   Functional      Prec-Agnostic or FQ
+ *              \       /
+ *               Concat
+ *                  |
+ *               Result
+ *
+ * For the above model, during scale factors propagation, when left Concat input
+ * will be selected as a scale factors source, then algorithm will not be able to
+ * apply the scale factors to the right Concat input and will throw exception.
+ * InsertIdentityForPrecAgnosticConcatInput pass adds Identity layer, which
+ * is capable of storing scale factors, so the scale factors propagation can proceed:
+ *
+ *                    Parameter
+ *                        |
+ *                  Prec-Agnostic or FQ
+ *                        |
+ *      Functional     Identity
+ *              \       /
+ *               Concat
+ *                  |
+ *               Result
+ */
+class InsertIdentityForPrecAgnosticConcatInput : public ov::pass::ModelPass {
+    /**
+     * @brief Check if FakeQuantize exists on any input
+     */
+    bool has_fq_on_any_input(const std::shared_ptr<ov::Node> concat_node);
+
+    /**
+     * @brief Check if at least two inputs are not identical
+     */
+    bool all_inputs_point_the_same_node(const std::shared_ptr<Node>& node);
+
+    /**
+     * @brief Return vector of nodes for Identity insertion
+     */
+    std::vector<std::shared_ptr<ov::Node>> get_nodes_for_identity_insertion(
+        const std::shared_ptr<ov::Node>& concat_node);
+
+    /**
+     * @brief Invokes Identity layer insertion after each node in vector
+     */
+    bool insert_identity_after_nodes(const std::vector<std::shared_ptr<ov::Node>>& nodes,
+                                     const std::shared_ptr<ov::Node>& next);
+
+    /**
+     * @brief Invoke Identity layer insertion in case the Concat input is unable
+     * to store scale factors
+     */
+    bool insert_identity_for_prec_agnostic_concat_inputs(const std::shared_ptr<ov::Node>& node);
+
+    /**
+     * @brief Find the output index of 'prev' layer, on which it is connected to 'next' layer
+     */
+    size_t find_prev_layer_output_index(std::shared_ptr<ov::Node> prev, std::shared_ptr<ov::Node> next);
```

Comment:
in previous declarations const & are used for shared_ptr-s, but in this method not. Is it done on purpose?

Reply:
No. Changed.